### PR TITLE
Rains dig man page and implementation update

### LINKED ISSUE #177: Update rainsdig and its man page to match the current protocol draft
rainsdig's man page and implementation our out of sync with each other and also with the new protocol specification.
Use spf13/pflag instead of std/flag for better usability

### PR DESCRIPTION
**Pull request by fehlmach**
This PR updates the man page of rdig and the implementation to match the updated RAINS draft.
Each query option can be set by an individual flag.
Fixes #177

---
This change is [Reviewable](https://reviewable.io/reviews/netsec-ethz/rains/176)

### CODE REVIEW

File: docs/man/rdig.md [hunk 1]
```diff
@@ -3,44 +3,75 @@ rdig(1) -- A RAINS query tool
...
 
-rdig (short for RAINS dig) is a tool for querying RAINS servers from the command line. It performs lookups of the provided domain names and prints the results on the command line.
+rdig (short for RAINS dig) is a tool for querying RAINS servers from the command line. It performs
+lookups of the provided domain names and prints the results on the command line in zone file format.
 
-The queries are sent to the resolvers specified in the */etc/rains.conf* file, unless it is told to use a specific server via command line arguments.
+## Simple Usage
 
-When no command line arguments are specified, then 
+A typical usage of dig looks like:
```
Comment (rayhaanj) [inline docs/man/rdig.md]: rdig or dig?

File: docs/man/rdig.md [hunk 2]
```diff
@@ -3,44 +3,75 @@ rdig(1) -- A RAINS query tool
...
 
-## OPTIONS
+dig @server name type
+
+where:
+* server: is the IP address or a name of the RAINS server to which the query will be sent. The
+    address can be an IPv4 address in dotted-decimal notation or an IPv6 address in colon-delimited
+    notation. If the provided argument is a host name, rdig first resolves the IP address of that
+    host before sending the actual query to this RAINS server. If no server argument is provided,
+    dig consults /etc/resolv.conf and queries the name servers listed there.
```
Comment (rayhaanj) [inline docs/man/rdig.md]: Not yet looking in resolv.conf but I guess this will be coming soon

Comment (fehlmach) [reply docs/man/rdig.md]: I filed an issue #178 for that. @juagargi and @britram should this be part of RAINS for SCIONlab milestone?

File: cmd/rdig/rdig.go [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+	case 0:
+		log.Fatal("Error: no domain name specified.")
+	case 1:
+		name = flag.Arg(0)
+		if strings.HasPrefix(name, "@") {
+			log.Fatal("Error: no domain name specified.")
+		}
+	case 2:
+		server = flag.Arg(0)
+		if !strings.HasPrefix(server, "@") {
```
[the remark this answers is not in the document]
Comment (fehlmach) [reply cmd/rdig/rdig.go]: done

**Pull request merged**